tpp-rt/XsmmRunnerUtils.cpp

```diff
@@ -344,62 +287,44 @@ extern "C" void _mlir_ciface_matrix_copy_NC_to_NCNC(
   matrix_copy_NC_to_NCNC(addr_input, addr_output, 1, C, N, c, n);
 }
 
-extern "C" void _mlir_ciface_xsmm_brgemm_invoke_f32(
-    int64_t addr, UnrankedMemRefType<float> *A, UnrankedMemRefType<float> *B,
-    UnrankedMemRefType<float> *C, int64_t numBatches) {
+extern "C" void _mlir_ciface_xsmm_brgemm_invoke(const libxsmm_datatype dType,
+                                                int64_t addr,
+                                                UnrankedMemRefType<char> *A,
```

Comment:
Not an expert here but would void work instead of char?

Reply:
problem is dynamicmemreftype does not support void type, which we need to get the data pointer and the offset.